src/draft-kowalik-rpp-data-objects.md

```diff
@@ -55,13 +55,34 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 
 # Resource Definition Principles
 
+## Primitive Data Types
+
+RPP data elements use strict typing, meaning that each element must conform exactly to its declared primitive data type, and type violations MUST be treated as errors by implementations. The exact specifications for these types, including allowed ranges, encoding, and formatting, are determined by the representation format used (e.g., JSON, XML, CBOR). New RPP non-primitive data types based on existing primitive data types MAY be defined to support additional features.
+
+Data Types table:
+| Type       | Description                                           | Example                 |
+| ---------- | ----------------------------------------------------- | ----------------------- |
+| `String`   | Sequence of Unicode characters.                       | `"host.example"`         |
+| `Integer`  | Whole number, positive or negative.                   | `42`                    |
+| `Boolean`  | Logical true or false value.                          | `true`, `false`         |
+| `Float`    | Decimal number, supporting fractional values.         | `3.14159`               |
+| `Date`     | Calendar date in YYYY-MM-DD format.                   | `2025-10-27`            |
+| `Timestamp` | Timestamp (date and time) in [@!RFC3339] format. | `2025-10-27T09:42:51Z`  |
+| `URI`      | Uniform Resource Identifier.                          | `"https://host.example"` |
+| `Binary`   | Raw binary data, encoded as base64 or hexadecimal.    | `"UlBQIFNheXMgSGk="`      |
+
+A> TODO: do we need a Date type separately from Timestamp?
+
+Timestamp (Date and time attribute) values MUST be represented in Universal Coordinated Time (UTC) using the Gregorian calendar using date-time form as defined in [@!RFC3339]. In EPP Compatibility Profile upper case "T" and "Z" characters MUST be used.
```

Comment:
We don't have to do it separately. but then maybe the table is not enough to comprehend all the necessary information and a section for each type would be better?